notebooks/circuits_with_bloqade.py

```diff
@@ -0,0 +1,763 @@
+# %%
+from bloqade import squin
+import numpy as np
+
+# Some types we will use, useful for type hints
+from kirin.dialects.ilist import IList
+import bloqade.types
+from typing import Any
+from kirin.types import Method
+Register = IList[bloqade.types.Qubit, Any]
+
+# %% [markdown]
+# # Circuits with Bloqade
+# 
+# In this tutorial we will demonstrate how to write circuits and quantum executions with Bloqade. Specifically, we will use the `squin` dialect set from the compiler toolchain `kirin`. `SQUIN` stands for `S`tructural `Qu`antum `IN`struction set and is the circuit-level representation of quantum executions. It is built on top of the `kirin` framework, an [open-source compiler toolchain](https://queracomputing.github.io/kirin/latest/) for embedded domain-specific languages (eDSLs) that target scientific computing kernels. A key feature of squin is the _kernel_, which can roughly be seen as the object which will be executed on the target hardware. Naturally, this hardware could be a quantum computer but it also extends to classical execution as well, such as mid-circuit feedforward or even non-quantum execution such as robotics.
+# 
+# These kernels can be built using decorators of python functions. We will use the `@squin.kernel` decorator in this notebook but keep in mind that other eDSLs have different decorators inherited from base Kirin decorators. The decorator lowers the python AST into a kirin SSA (single static assignment) form, which is a useful intermediate representation for compiler analysis. You don't have to worry too much about SSA or compilers very much here, but if you want to learn more check out the [kirin documentation](https://queracomputing.github.io/kirin/latest/).
+
+# %%
+try:
+    del hello_world
+except:
+    pass
+@squin.kernel
+def hello_world(theta:float) -> IList[bool, Any]:
+    """
+    Prepare a Bell state and measure in a basis that might have a Bell violation
+    """
+    qubits = squin.qubit.new(2)
+    squin.gate.h(qubits[0])
+    squin.gate.cx(qubits[0], qubits[1])
+    squin.gate.rx(theta, qubits[0])
+    bits = squin.qubit.measure(qubits)
+    return bits
+hello_world.print()
```

Comment:
Since you're mention some compiler basics above, it might make sense to (briefly) explain what `.print()` does and what an IR is.